saleor/graphql/order/types.py

```diff
@@ -615,6 +626,34 @@ def _resolve_stock(fulfillment_lines: List[models.FulfillmentLine]):
             .then(_resolve_stock)
         )
 
+    @staticmethod
+    def resolve_shipping_refunded_amount(root: models.Fulfillment, info):
+        if root.shipping_refund_amount is None:
+            return None
+
+        def _resolve_shipping_refund(order):
+            return prices.Money(root.shipping_refund_amount, currency=order.currency)
+
+        return (
+            OrderByIdLoader(info.context)
+            .load(root.order_id)
+            .then(_resolve_shipping_refund)
+        )
+
+    @staticmethod
+    def resolve_total_refunded_amount(root: models.Fulfillment, info):
+        if root.total_refund_amount is None:
+            return None
+
+        def _resolve_total_refund_amount(order):
+            return prices.Money(root.total_refund_amount, currency=order.currency)
```

Comment:
Shouldn't we add `currency` on `Fulfillment` model instead fetching it from order each time? (I mean on `main` branch, I don't suggest to introduce it from 3.14)

Reply:
Honestly, these changes are only to support the old payment flow & refund on webhook subscriptions. We will not need it when we will drop Payment API and fully move to transactionAPI. So I would say, that it doesn't make sense ta add this to Fulfillment model.